### PR TITLE
feat(RAIN-91334): allow Rego queries to be created and retrieved

### PR DESCRIPTION
**Pull request by incredibler**
## Summary

Allow a Rego query to be created from a given yaml file with lacework query create -f MyRego.yaml, and it can be returned via lacework query show command.  The yaml file have three fields:
```
---
queryId: TestRegoQueryViaCli2
queryLanguage: Rego
queryText: |-
  package lacework.policies.lacework_global_134
  import future.keywords
  import data.lacework
  import data.lacework.assessment
  source := lacework.aws.cfg.list("s3", "list-buckets")
  assess := assessment.violation(input, "just because")
```
Now the input and output of query creation need to have a new field `queryLanguage` which is optional.

## How did you test this change?
* Run cli locally in GoLand with program arguments `query create -f rego_query.yaml`,  got response:
```
The query TestRegoQueryViaCli2 was deleted.
```
* Run the same command again with an invalid queryLanguage, got a failure:
```
ERROR unable to create query: 
  [POST] https://dev8.dev8.corp.lacework.net/api/v2/Queries
  [400] Problem parsing JSON, the value for key queryLanguage is of the wrong type
```
* Run cli locally in Goland with program arguments  `query show TestRegoQueryViaCli2 --json`,  got response as expected:
```
{
  "lastUpdateTime": "2023-11-29T06:48:29.000Z",
  "lastUpdateUser": "kun.chen@lacework.net",
  "owner": "kun.chen@lacework.net",
  "queryId": "TestRegoQueryViaCli2",
  "queryLanguage": "Rego",
  "queryText": "package lacework.policies.lacework_global_134\nimport future.keywords\nimport data.lacework\nimport data.lacework.assessment\nsource := lacework.aws.cfg.list(\"s3\", \"list-buckets\")\nassess := assessment.violation(input, \"just because\")",
  "resultSchema": null
}

```

## Issue

https://lacework.atlassian.net/browse/RAIN-91334

### CODE REVIEW

**Review by jon-stewart — APPROVED** [review]

**Review by qdfuxiao — APPROVED** [review]

File: api/lql_test.go [hunk 1]
```diff
@@ -46,6 +47,25 @@ var (
 queryId: %s
 queryText: %s`, newQuery.QueryID, newQuery.QueryText)
 	lqlErrorReponse = `{ "message": "This is an error message" }`
+
+	regoQueryID      = "my_rego"
+	newRegoQueryText = "package lacework.policies.lacework_global_134\n" +
+		"import future.keywords\n" +
+		"import data.lacework\n" +
+		"import data.lacework.assessment\n" +
```
Comment (gspofford-lw) [inline api/lql_test.go:55]: I'd recommend we not use this import.

File: api/lql.go [hunk 2]
```diff
@@ -59,6 +61,7 @@ type UpdateQuery struct {
 
 type Query struct {
 	QueryID        string                   `json:"queryId" yaml:"queryId"`
+	QueryLanguage  string                   `json:"queryLanguage,omitempty" yaml:"queryLanguage,omitempty"`
```
Comment (gspofford-lw) [inline api/lql.go:64]: Needs to be `*string` to represent the actual API.

File: api/lql_test.go [hunk 3]
```diff
@@ -225,6 +261,22 @@ func TestQueryGetQueryByIDOK(t *testing.T) {
 	assert.Nil(t, err)
 
 	assert.Equal(t, getExpected, getActual)
+
+	if strings.Contains(expectedResponseData, "queryLanguage") {
```
Comment (gspofford-lw) [inline api/lql_test.go:265]: If we are expecting a queryLanguage in the response, should we be validating that it's the right one?

File: api/lql_test.go [hunk 4]
```diff
@@ -151,10 +171,26 @@ func TestQueryCreateOK(t *testing.T) {
 	_ = json.Unmarshal([]byte(mockResponse), &createExpected)
 
 	var createActual api.QueryResponse
-	createActual, err = c.V2.Query.Create(newQuery)
+	createActual, err = c.V2.Query.Create(testQuery)
 	assert.Nil(t, err)
 
 	assert.Equal(t, createExpected, createActual)
+
+	if strings.Contains(expectedResponseData, "queryLanguage") {
```
Comment (gspofford-lw) [inline api/lql_test.go:179]: If we are expecting a queryLanguage in the response, should we be validating that it's the right one?

File: api/lql.go [hunk 5]
```diff
@@ -29,8 +29,9 @@ import (
 )
 
 type NewQuery struct {
-	QueryID   string `json:"queryId" yaml:"queryId"`
-	QueryText string `json:"queryText" yaml:"queryText"`
+	QueryID       string `json:"queryId" yaml:"queryId"`
+	QueryLanguage string `json:"queryLanguage,omitempty" yaml:"queryLanguage,omitempty"`
```
Comment (gspofford-lw) [inline api/lql.go:33]: This also should be `*string`, yes? Otherwise we might submit the `""` string.

**Pull request merged**